app/services/pipeline_report_service.rb

```diff
@@ -103,14 +112,27 @@ def generate
     ]
 
     tax_ids = counts_by_tax_level[TaxonCount::TAX_LEVEL_GENUS].keys
+    # If a species has an undefined genus (id < 0), the TaxonLineage id is based off the
+    # species id rather than genus id, so select those species ids as well.
+    species_with_missing_genus = []
```

Comment:
Did you notice the impact of this?

Reply:
Let's add another todo here to clear this up once we clean up the data.

Reply:
This definitely slows things down. I eyeballed the times for the `fetch_taxon_lineage` for a sample, and it goes from roughly 0.4 to 0.6.